contracts/core/OriginCore.sol

```diff
@@ -115,25 +57,60 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
      * Mapping of block hashes to block headers that were reported with the
      * respective hash.
      */
-    mapping (bytes32 => Header) public reportedHeaders;
+    mapping (bytes32 => MetaBlock.Header) public reportedHeaders;
+
+    /**
+     * Mapping of kernel hash to transition object map.
+     * where transition object map is transition hash to transition mapping.
+     */
+    mapping(bytes32 => mapping(bytes32 => MetaBlock.AuxiliaryTransition)) public proposedMetaBlock;
 
     /* Constructor */
 
     /**
-     * @param _chainIdAuxiliary The id of the auxiliary chain that this core
-     *                          contract tracks.
+     * @param _auxiliaryCoreIdentifier The core identifier of the auxiliary
+     *                        chain that this core contract tracks.
      * @param _ost The address of the OST ERC-20 token.
+     * @param _initialValidators The array of addresses of the validators.
+     * @param _validatorsWeights The array of weights that corresponds to
+     *                        the updated validators.
+     * @param _initialAuxiliaryGas Initial gas consumed on auxiliary before
+     *                             reporting genesis meta-block.
+     * @param _initialTransactionRoot Transaction root of auxiliary chain
+     *                                before reporting genesis meta-block.
      */
     constructor(
-        uint256 _chainIdAuxiliary,
-        address _ost
+        bytes32 _auxiliaryCoreIdentifier,
+        address _ost,
+        address[] _initialValidators,
+        uint256[] _validatorsWeights,
```

Comment:
The core should not need to know about initial validators. The validators should be tracked in the Stake contract only (for origin). It is much more complex to deploy Stake (see #400).

It is probably best to only pass the address of the stake contract here.

The Genesis meta-block does not need to contain the initial set of validators.